File: Algoritmos Traduzidos/Perl/Algoritmos_Filipe_Dias/106.pl
print "Este programa calcula N factorial\n";
print "Digite o valor para se calculado:\n";
$num = <STDIN>;
if ($num >= 0)
{
	$fat = 1;
	$n = $num;
	while($n > 0)
	{
		$fat = $fat * $n;
		$n = $n - 1;
	}
	print "O numero factorial de $num e $fat\n";
}
else
{
	print "O calculo do factorial aplica-se somente aos numeros naturais\n";
}

File: Algoritmos Traduzidos/Ruby/Algoritmos_Filipe_Dias/106.rb
puts "Este programa calcula N factorial"
puts "Digite o valor para ser calculado:"
num = gets
if num.to_i >= 0
  fat = 1
  n = num.to_i
  while n.to_i > 0
    fat = fat.to_i * n.to_i
    n = n.to_i - 1
  end
  puts "O numero factorial de #{num} e #{fat}"
else
    puts "O calculo do factorial aplica-se somente aos numeros naturais"
end